File: problems/14/Solution.swift
class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        if strs.count == 0 {
            return ""
        }

        for i in 0..<strs[0].characters.count {
            let index = strs[0].index(strs[0].startIndex, offsetBy: i)
            let target = strs[0][index]
            for j in 1..<strs.count {
                if i == strs[j].characters.count || strs[j][index] != target {
                    return strs[0].substring(to: index)
                }
            }

        }

        return strs[0]
    }
}

File: problems/14/Solution.cpp
// 14. Longest Common Prefix

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string answer = "";
        int minLength = strs[0].length();
        for (int i = 1; i < strs.size(); i++) {
            if (strs[i].length() < minLength) {
                minLength = strs[i].length();
            }
        }
        for (int j = 0; j < minLength; j++) {
            auto c = strs[0][j];
            for (int i = 1; i < strs.size(); i++) {
                if (c != strs[i][j]) {
                    return answer;
                }
            }
            answer += c;
        }
        return answer;
    }
};
